dragonchain/job_processor/job_processor.py

```diff
@@ -97,6 +108,7 @@ def start_task() -> None:
         if job and (job.status.succeeded or job.status.failed):
             delete_existing_job(task_definition)
         attempt_job_launch(task_definition)
+        redis.lpop_sync(PENDING_TASK_KEY)
```

Comment:
should pop this off when it does the explicit `return` too, since in that case we're _trying_ to throw away the task; we don't want it coming back